File: 1-yahtzee-java/src/main/java/yahtzee/Category.java
package yahtzee;

import java.util.Map;

public enum Category {
    Ones(1),
    Twos(2),
    Threes(3);

    private final int value;

    Category(int value) {
        this.value = value;
    }

    public int computeScore(Map<Dice, Integer> dice) {
        int score = 0;
        for (Integer value : dice.values()) {
            if (value.equals(this.value)) {
                score++;
            }
        }
        return score;
    }
}

File: 2-yahtzee-php/src/Yahtzee/Category.php
<?php


namespace Yahtzee;


class Category
{
    /** @var string */
    private $label;

    /** @var int */
    private $value;

    /**
     * @param string $label
     * @param int $value
     */
    private function __construct($label, $value)
    {
        $this->label = $label;
        $this->value = $value;
    }

    /**
     * @return Category
     */
    public static function Ones()
    {
        return new self('Ones', 1);
    }

    /**
     * @return Category
     */
    public static function Twos()
    {
        return new self('Twos', 2);
    }

    /**
     * @return Category
     */
    public static function Threes()
    {
        return new self('Threes', 3);
    }

    /**
     * @return Category[]
     */
    public static function all()
    {
        return [
            self::Ones(),
            self::Twos(),
            self::Threes()
        ];
    }

    /**
     * @return string
     */
    function __toString()
    {
        return $this->label;
    }

    /**
     * @param array $dice
     * @return int
     */
    public function calculateScore($dice)
    {
        return count(array_filter($dice, function ($die) {
            return $die == $this->value;
        }));
    }
}